app/src/main/java/sw19/moring03/paint/MainActivity.java

```
@@ -99,11 +127,30 @@ public void chooseNewTool(View view) {
             case R.id.drawPathButton:
                 setChosenTool(Tool.DRAW_PATH);
                 break;
+            case R.id.takePhoto:
+                pickFromGallery();
+                break;
         }
 
         toolChooserMenu.dismiss();
     }
 
+    public void onActivityResult(int requestCode,int resultCode,Intent data){
+
+        // Result code is RESULT_OK only if the user selects an Image
+        if (resultCode == Activity.RESULT_OK)
+            switch (requestCode){
+                case PICK_IMAGE:
+                    Uri imageUri = data.getData();
+                    try {
+                        Bitmap bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), imageUri);
+                        new_photo = bitmap;
+                        setChosenTool(Tool.TAKE_PHOTO);
+                    }catch(Exception ex){}
```

no empty catch block